### PR TITLE
feat: Add option to sort alarms

### LINKED ISSUE #193: Alarms Not in Chronological Order
Alarms are not presenting in Chronological Order in the menu. As someone that uses alarms for everything, this is more of an inconvenience when hunting down a previous alarm to use again.

### LINKED ISSUE #215: [suggestion] option to sort alarms by time
hi,
currently all alarms are sorted by date of creation.
an option which sorts them by the time they ring (as it is default for samsung and miui) whould be very nice!

### PR DESCRIPTION
**Pull request by jkemming**
I have implemented an option to sort alarms either by their creation time or the time of day they ring at. Design and implementation are similar to how it's done in File Manager Pro, so I hope it's okay in terms of UX.

Please let me know if I made any errors as this is (hopefully) my first contribution here.

This PR should close #193 and close #215

### CODE REVIEW

**Comment by JamesDean829** [comment]
Thanks, this is quite useful!

File: app/src/main/res/values-az/strings.xml [hunk 1]
```diff
@@ -56,6 +56,8 @@
         <b>Reddit:</b>
         https://www.reddit.com/r/SimpleMobileTools
     </string>
+    <string name="created_at">Created at</string>
```
Comment (tibbi) [inline app/src/main/res/values-az/strings.xml:59]: move the new strings into some proper place, not at the bottom

Comment (jkemming) [reply app/src/main/res/values-az/strings.xml]: Thanks for reviewing, I moved the new strings up to a different place. I also noticed some unnecessary leftover imports from my debugging, which I fixed as well. I would be glad if you could take another look.

File: app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt [hunk 2]
```diff
@@ -72,13 +90,14 @@ class AlarmFragment : Fragment(), ToggleAlarmInterface {
...
-        if (context?.getNextAlarm()?.isEmpty() == true) {
+        val context = context ?: return
+        alarms = context.dbHelper.getAlarms(context.config.alarmSort)
+        if (context.getNextAlarm().isEmpty()) {
             alarms.forEach {
                 if (it.days == TODAY_BIT && it.isEnabled && it.timeInMinutes <= getCurrentDayMinutes()) {
                     it.isEnabled = false
                     ensureBackgroundThread {
-                        context?.dbHelper?.updateAlarmEnabledState(it.id, false)
+                        context.dbHelper.updateAlarmEnabledState(it.id, false)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt:100]: this run on a background thread, context could be null by then. Avoid changing unrelated things please, it is done like this for a reason.

Comment (jkemming) [reply app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt:100]: That is still checked, in line 93 the function returns if `context` is `null` (the same behavior as before). I thought this change would make it a bit easier to read, but if you disagree I can change it back and just add the necessary sorting.

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt:100]: you check it above, but like 100 is already on a background thread and context could become null meanwhile

Comment (jkemming) [reply app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt]: You are right, I'm sorry I didn't realize that. I reverted the changes there now and just left the sort parameter itself in, I think that should cover it.

**Comment by tibbi** [comment]
seems to be good, thanks :) Ill just tweak a few things

**Comment by tibbi** [comment]
actually Ill tweak a bit more than I wanted, you seriously overengineered it. It needs to be done a lot simpler

**Pull request merged**